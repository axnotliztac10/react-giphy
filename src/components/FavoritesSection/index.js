import React from 'react';

import GiphyCard from 'components/GiphyCard';
import { Text, Box, Flex } from 'rebass';

export default class FavoritesSection extends React.PureComponent {
  renderFavorite = (favorite, index) => {
    const { onAddRemoveFavorite } = this.props;

    return (
      <GiphyCard
        giphy={favorite}
        key= {index}
        onAddRemoveFavorite={onAddRemoveFavorite}
      />
    );
  }

  render() {
    const { favorites } = this.props;

    return (
      <Box my={4}>
        <Text textAlign="center" fontSize={25}>My favorites</Text>
        <Flex
          flexDirection="row"
          justifyContent="center"
          alignItems="flex-start"
          flexWrap="wrap"
        >
          {favorites.map(this.renderFavorite)}
        </Flex>
      </Box>
    );
  }
}
