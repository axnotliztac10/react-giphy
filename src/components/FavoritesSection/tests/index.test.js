import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import FavoritesSection from '../index';

describe('<FavoritesSection />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <FavoritesSection
        favorites={[{ hideOnSearchFavorites: true }, { hideOnSearchFavorites: false }]}
        onAddRemoveFavorite={() => {}}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
