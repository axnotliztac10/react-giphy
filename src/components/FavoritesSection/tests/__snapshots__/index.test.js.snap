// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesSection /> should render correctly 1`] = `
<styled.div
  my={4}
>
  <Styled(styled.div)
    fontSize={25}
    textAlign="center"
  >
    My favorites
  </Styled(styled.div)>
  <Styled(styled.div)
    alignItems="flex-start"
    flexDirection="row"
    flexWrap="wrap"
    justifyContent="center"
  >
    <GiphyCard
      giphy={
        Object {
          "hideOnSearchFavorites": false,
        }
      }
      key="1"
      onAddRemoveFavorite={[Function]}
    />
  </Styled(styled.div)>
</styled.div>
`;
